apps/aehttp/test/aehttp_integration_SUITE.erl

```diff
@@ -817,20 +817,36 @@ internal_block_pending(_Config) ->
                 <<"BlockWithTxsHashes">>,
                 block_to_endpoint_map(PendingBlock)),
     ct:log("Expected pending block ~p", [ExpectedPendingTx]),
-    {ok, 200, PendingTxDefault} = get_internal_block_preset("pending",
-                                                             default),
-    {ok, 200, PendingTxHashes} = get_internal_block_preset("pending",
-                                                             false),
-    true = equal_block_maps(ExpectedPendingTx, PendingTxDefault), 
-    true = equal_block_maps(ExpectedPendingTx, PendingTxHashes), 
+    GetPending =
+        fun TryToGet(_Opt, Attempts) when Attempts < 1 ->
+                {error, no_pending_block_returned};
+            TryToGet(Opt, Attempts) ->
+                case get_internal_block_preset("pending", Opt) of
+                    {ok, 200, P} -> {ok, P};
+                    {ok, 404, _} -> TryToGet(Opt, Attempts -1)
+                end
+        end,
+    {ok, PendingTxDefault} = GetPending(default, 10),
+    {ok, PendingTxHashes} = GetPending(false, 10),
+    ValidateKeys =
+        fun(Map1, Map2, Key) ->
+            true = maps:get(Key, Map1, not_found1) =:=
+                   maps:get(Key, Map2, not_found2)
+        end,
+    ValidateKeys(ExpectedPendingTx, PendingTxDefault, <<"prev_hash">>),
+    ValidateKeys(ExpectedPendingTx, PendingTxDefault, <<"data_schema">>),
+
+    ValidateKeys(ExpectedPendingTx, PendingTxHashes, <<"prev_hash">>),
```

Comment:
This should not have a different prev_hash (since it is not expected to mine a block durring that part of the test)

Reply:
> This should not have a different prev_hash (since it is not expected to mine a block durring that part of the test)

Maybe writing this as a comment here would help reader.